### ISSUE TITLE
Inline Google Maps support

### ISSUE THREAD

**Issue by FlutterIssues** [flutter/flutter#73]
<a href="https://github.com/Hixie"><img src="https://avatars.githubusercontent.com/u/551196?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [Hixie](https://github.com/Hixie)**
_Thursday Jul 16, 2015 at 22:26 GMT_
_Originally opened as https://github.com/flutter/engine/issues/30_

---

_From @collinjackson on July 8, 2015 20:23_

We are going to want to have support for maps in Sky.

There are a few ways we might go about doing this:
1. Use native maps widget for each platform to render the map
2. Use maps widget for each platform to render to a texture and draw the map ourselves
3. Develop a custom maps SDK for Sky

The second option would be ideal but may not be possible. We should investigate which approach is best and build a demo app.

_Copied from original issue: domokit/mojo#310_

**Comment by FlutterIssues** [comment]
<a href="https://github.com/eseidelGoogle"><img src="https://avatars.githubusercontent.com/u/11857803?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [eseidelGoogle](https://github.com/eseidelGoogle)**
_Wednesday Aug 05, 2015 at 19:35 GMT_

----

We might need this for the Fitness app, depending.

**Comment by Hixie** [comment]
This is presumably going to be a "third-party" component, right?

**Comment by collinjackson** [comment]
We'll almost certainly want to use some non-Dart SDK to provide Maps support, so yes.

**Comment by eseidelGoogle** [comment]
We expect developers will use Maps by using a "native" iOS or Android maps view, per:
https://docs.google.com/document/d/1DOfwpL6VojCAG_zjjcA_z5ekF7rTIohV8dWn2o7_IME/edit

**Comment by a14n** [comment]
I don't see in this doc descriptions of how to integrate native widget (like Maps) into your flutter apps. Have I missed something ?

**Comment by eseidelGoogle** [comment]
Sorry, I may have been a bit overzealous in my linking.  Said doc is pre-work for Flutter to participate as a normal View inside Obj-C/Java applications.  The path to add other Obj-C/Java code to them is not yet documented, no.

**Comment by tjerkw** [comment]
As a developer i really want to use the native Google Maps component to render inside my flutter app. I just created a flutter app and this is kind of a blocker. The only solution is to build it in Java/Android and open it when clicking a flutter button.

Ideally I would like a 'NativeView' in flutter, which renders a native widget inside flutter.
This is probably quite difficult because you have merge rendering / events into the flutter rendering engine.

It also goes against the flutter 'one widget for both platforms'. However, sometimes complex widgets (like a map) are really something you want to reuse.

**Comment by Goutte** [comment]
Hello ! I would love to know the roadmap of this feature, or even if one even exists. I have two mobile apps in the works and both rely heavily on a map. I love Dart, and I'm looking for an excuse to try Flutter.

The solution with `FlutterView` looks overly complicated, from the point of view of a rookie who has never written a single line of Flutter. The other solutions do not suit my need, I need a fully interactive map.

I don't know how hairy that is to implement in Flutter, nor if you even plan to implement it someday. Any info would be appreciated. I guess the resolution of this issue will be my signal to start development with Flutter.

**Comment by debunge** [comment]
Is there any chance maps will be implemented in 2017? I love Dart and would like very much to use Flutter, but I really need non-static maps, like showing points of interest nearby and showing the description upon the click on the point.

**Comment by jingweno** [comment]
Any updates on this?

**Comment by markmooibroek** [comment]
First of all I (fulltime Android dev) love flutter and the progress that is made the last months is impressive. That said, I think this issue is a pretty big blocker for a lot of people that are willing to migrate their business in the future. 

I'm currently doing a case study for the company i am working for and if a NativeView ([per tjerkw's suggestion](https://github.com/flutter/flutter/issues/73#issuecomment-245218748)) feature exists we could use that to properly bridge the gap between the different platforms!

**Comment by abarth** [comment]
> I think this issue is a pretty big blocker for a lot of people that are willing to migrate their business in the future.

Yes.  We've gotten that feedback consistently.  It's just a question of what implementation approach we want to take and when to schedule the work.

**Comment by bramvbilsen** [comment]
Is there any progress on this? I've got two apps sketched that require maps and would like to make them my first Flutter app in the stores. So is this going to come out within the next few months or are we talking next year? :)

**Comment by Hixie** [comment]
We haven't begun work specifically on maps, but we have begun studying inline video and we believe that this should pave the way for this work. We cannot give an ETA at this time.

**Comment by wmleler** [comment]
I too would like to be able to use maps in Flutter, but there is a simple case that might be a workaround. If I could have a full screen native map being displayed, and have a semi-transparent Flutter layer above it, and some easy way to communicate between the native map layer and the Flutter app (e.g., to tell when the map moves, its center lat/lng, zoom level, etc.) that would work for quite a few applications that require maps. In other words, no inline maps embedded in a Flutter view.

It would be even better if this is generalized so that a flutter layer can be composited on top of any Android layer -- not just maps, but 3D views (including VR), or video, etc.

**Comment by sethladd** [comment]
Found this plugin from @FaisalAbid : https://github.com/FaisalAbid/flutter-google-maps

**Comment by bramvbilsen** [comment]
@sethladd It's a great first step. But as Flutter focuses on cross platform development, it should have a plugin for both platforms. But I'm sure tons of new plugins will come when inline views are available.

**Comment by gabrielsallesrg** [comment]
I created a solution that works in Android and iOS and I made a video showing the result.
It is using Google Static Maps and I implemented the zoom in / zoom out features and added a pin in the user location, but it is possible to implement other features, like moving north/south/east/west and add a bunch of pins.
To make it more efficient, I stacked 3 maps - the one that you see, one with +1 level of zoom and one with -1 level of zoom. Because Flutter is awesome in caching images, when you press + or -, it will get the cached image, so it will pretty fast. 
```
            new Stack(
              children: <Widget>[
                new Center(
                  child: new CircularProgressIndicator(),
                ),
                //Zoom in Map Image for caching
                new Image.network(
                  mapUrl.zoomInMapUrl,
                  fit: BoxFit.contain,
                ),
                //Zoom out Map Image for caching
                new Image.network(
                  mapUrl.zoomOutMapUrl,
                  fit: BoxFit.contain,
                ),
                //Map Image
                new Image.network(
                  mapUrl.mapUrl,
                  fit: BoxFit.contain,
                  gaplessPlayback: true,
                ),
                new Positioned(
                  bottom: 16.0,
                  right: 16.0,
                  child: new Column(
                    children: <Widget>[
                      //Zoom in
                      new Container(
                        color: Colors.black54,
                        child: new IconButton(
                          icon: new Icon(Icons.add),
                          color: Colors.white,
                          onPressed: () => zoomInMap(true),
                        ),
                      ),
                      new Divider(),
                      //Zoom out
                      new Container(
                        color: Colors.black54,
                        child: new IconButton(
                          icon: new Icon(Icons.remove),
                          color: Colors.white,
                          onPressed: () => zoomInMap(false),
                        ),
                      ),
                    ],
                  ),
                ),
              ],
            ),
```

Even though I am not making available the source code right now, the implementation is pretty much straight forward.
Don't forget to read the video description.

https://www.youtube.com/watch?v=PR2MdUiepyw

**Comment by bramvbilsen** [comment]
@gabrielsallesrg I didn't even know Google Maps provided functionality for that. This is for sure a good solution for now. Thanks for sharing this!

**Comment by gabrielsallesrg** [comment]
@bramvbilsen Just to avoid confusion, Google Static Maps does not provide zoom in / zoom out functionality. What really happens is that when you clicked the +/- button, it requests a new image.

**Comment by bramvbilsen** [comment]
Got it. Thanks :)

**Comment by tjerkw** [comment]
I think for now, if you need these advanced widgets, flutter is a no go area.
Whenever you start a project in flutter, you must understand that some designs that your designers come up with are not easy to implement.

The best workaround is a google static map, with a link to a fullscreen map component. Since a lot of apps are doing this now, it gives you a solution to build a flutter app with native like experience.

In the long term, i hope flutter has its own native gmaps widget. But before that happens flutter must be announced oficially by Google, and supporting it. Its unclear what the future of flutter now.

**Comment by bramvbilsen** [comment]
@tjerkw I do agree that it's still in alpha, so we can't expect every functionality from more mature frameworks yet. But hasn't Flutter been announced by Google yet? They even showed it at IO

**Comment by fredericletellier** [comment]
@tjerkw Very clear explanation for Gmaps.

I have already work with  VTM (vector-tile map) library for Android (opensciencemap/openstreetmap).  Is there a library adapted to flutter to manage an open-source map ?

**Comment by sethladd** [comment]
> But as Flutter focuses on cross platform development, it should have a plugin for both platforms

Agreed, but our ecosystem is only now booting up. We hope by sharing existing plugins we find, we can get more contributors to help author more features and functionality.

**Comment by GeekRishabh** [comment]
@tjerkw  can you share the code base please

**Comment by TitikshaDaga** [comment]
Is there any update on the Google maps integration with flutter?

**Comment by TitikshaDaga** [comment]
How do I add native maps in each platform and render the map?

**Comment by najeira** [comment]
@TitikshaDaga Inline Google Map is not yet supported.

You can show native Activity/ViewController with Google Map by plugin mechanism.

**Comment by TitikshaDaga** [comment]
https://github.com/FaisalAbid/flutter-google-maps/issues/1
Can any1 resolve this??

**Comment by timotheecour** [comment]
> You can show native Activity/ViewController with Google Map by plugin mechanism.

@najeira do you have sample code/pseudocode for this? Does that involve using a FlutterView?

**Comment by najeira** [comment]
@timotheecour [image_picker](https://github.com/flutter/plugins/tree/master/packages/image_picker) and [flutter_webview_plugin](https://github.com/dart-flitter/flutter_webview_plugin) are good example for using native Activity/ViewController.

**Comment by eseidelGoogle** [comment]
@mehmetf explained to me today that customer: mulligan does need maps, but believes they can solve their use cases with static map images for now, hence the customer tagging.

**Comment by johnpryan** [comment]
Shameless plug for those looking for a full-screen map solution: 

https://pub.dartlang.org/packages/map_view

**Comment by matteing** [comment]
Any news on this? This is a show stopper for geographic applications.

**Comment by wmleler** [comment]
This is being worked on now, and progress is being made.

**Comment by hundeva** [comment]
Any sneak peek? 😃

**Comment by theobouwman** [comment]
> This is being worked on now, and progress is being made.

@wmleler I'd love to see a working map...

**Comment by wmleler** [comment]
@theobouwman Right now, this is the best solution -- https://pub.dartlang.org/packages/map_view

If you want to have a map act like it is part of a Flutter widget, you can insert a map as an image (using the static maps API -- https://developers.google.com/maps/documentation/static-maps/ ). When the user clicks on this static map image, open up a full page map, along with a widget to close this. After the user closes the full page map, replace the map image with an updated static map (showing changes in map zoom and scroll, any added markets, etc.)

**Comment by mclark4386** [comment]
@wmleler I think @theobouwman more meant he was looking forward to seeing the "native widget" integration. I'm excitedly waiting as well^^

**Comment by theobouwman** [comment]
@mclark4386 that's right.

**Comment by wmleler** [comment]
I would love to hear your use cases for what you call "native widget" integration. I've built quite a few apps using maps, and to me it is sufficient to have the ability to overlay flutter widgets on top of a full page map, and to have a static map embedded in a Flutter widget that opens up to a full page map. I'm not saying we shouldn't provide native widget integration, I'm just curious why people want it and I want to understand.

**Comment by bramvbilsen** [comment]
@wmleler There are a lot of use cases! For example an interactive map with restaurants nearby. You'd want to use an inline map with some custom layout on top of it to make your app look unique.

**Comment by matteing** [comment]
It's pretty basic stuff for geographic applications. Like, for example, an app that is a list of pins and GeoJSON meshes.

On Jan 3, 2018 1:09 PM, Wm Leler <notifications@github.com> wrote:

I would love to hear your use cases for what you call "native widget" integration. I've built quite a few apps using maps, and to me it is sufficient to have the ability to overlay flutter widgets on top of a full page map, and to have a static map embedded in a Flutter widget that opens up to a full page map. I'm not saying we shouldn't provide native widget integration, I'm just curious why people want it and I want to understand.

**Comment by Janamou** [comment]
@wmleler Our apps usually look like this. We have **tabs**, on the first tab is a google map. On the second is a listview. I am able to click on any pin on the map and then I see something like "partial" detail/info of that item from which I can launch navigation from the current position to that item (selected pin).

**Comment by sanketsahusoft** [comment]
@wmleler Please let me know if we have updates on this. We would love to start using Flutter but this is sort of a roadblock for us at the moment.

We are looking to build a location based service like Uber!

**Comment by matteing** [comment]
Same situation. Would definitely love updates on this.

On Jan 9, 2018 4:51 AM, Sanket Sahu <notifications@github.com> wrote:

@wmleler<https://github.com/wmleler> Please let me know if we have updates on this. We would love to start using Flutter but this is sort of a roadblock for us at the moment.

We are looking to build a location based service like Uber!

**Comment by wmleler** [comment]
Just so you aren't left hanging, we are aware that this is very important and we are working on it.

**Comment by ljfreelancer88** [comment]
I do have also location based project. Can' wait :(

**Comment by slightfoot** [comment]
@wmleler we are looking to port an application with a split Android/iOS code base over to Flutter. But the main feature uses Google Maps to display pins. So we are really waiting on this feature.

I was thinking about implementation and wondering if this could be done similar to the video support. Something along the lines of a FlutterNativeHostView which hosts the platform view and stubs out the child rendering process to a texture supplied by the Flutter framework. This could then be displayed like the video player in the Flutter view hierarchy. [Touch] Input would have to be passed over a method call to the plugin, but it would be doable.

Frankly the better way.. and the only other other way I can see it implemented, would be to rewrite/redeploy the Google Maps library in Dart. I would suppose this would take more time.

@wmleler I'd love to hear your thoughts on something like the host view. Perhaps this is a good direction for Flutter so that more libraries can be ported over by wrapping their current components in a Plugin.

**Comment by najeira** [comment]
You can show native Google Maps View over FlutterView.

The AdMob plugin is doing something similar:
https://github.com/flutter/plugins/blob/master/packages/firebase_admob/android/src/main/java/io/flutter/plugins/firebaseadmob/MobileAd.java#L242

**Comment by mravn-google** [comment]
@slightfoot The idea of a FlutterNativeHostView sounds interesting, though I am not sure how that would work technically on Android. Do you have a concrete approach in mind concerning how to "stub out the child rendering process"?

On iOS, we are currently working on a solution that integrates CALayers into the way the Flutter Engine does compositing. That should allow us to consume any UIView.

**Comment by NathanaelA** [comment]
I am fairly new to the Flutter eco-system, but if I understand how it works correctly; Skia is using OpenGL to do its rendering.   You might look at Google GVRKit on how they are doing it.   GVRUIRenderer can take a normal UI element and put it into the OpenGL scene.

**Comment by mravn-google** [comment]
@NathanaelA Thanks for the tip! Are you referring to https://github.com/googlevr/gvr-ios-sdk/blob/master/Samples/GVRKit/GVRUIViewRenderer.h?

/cc @sigurdm

**Comment by NathanaelA** [comment]
Yep, actually it is the .mm file: https://github.com/googlevr/gvr-ios-sdk/blob/master/Samples/GVRKit/GVRUIViewRenderer.mm  -- I'm using it currently to display a whole UIView based scene in my GVRKit project.

**Comment by franzsilva** [comment]
Any updates on this Feature?, have a client on the doorstep wanting a geolocated app like uber. and this feature would be really necessary. 

Thanks!!!

**Comment by mravn-google** [comment]
We are actively working on it. Landing our solution is still a few weeks away. In the meantime you may want to use [AppTree's plugin](https://pub.dartlang.org/packages/map_view). While that plugin offers only full-screen interactive maps and inline static ones, it might be enough for you to get a prototype up and running.

It should be noted that GoogleMaps on Android is not currently open to integration into custom, OpenGL compositing like the Flutter engine does. So while we will be able to deliver interactive, inline maps also on Android, they will be restricted for now so that they are either top-most on the z axis or non-interactive.

In other words, it will take some time before Flutter is able to draw a button on top of an interactive map on Android. You can still draw such buttons with custom Java/Kotlin code of course. We are working with the maps team to resolve that issue too, but it will take some time, as an API addition seems necessary.

**Comment by wmleler** [comment]
By the way, AppTree, in addition to their plugin that Mikkel mentioned, is
currently working on a new package (not a plugin!) that implements maps in
Flutter directly (the basemaps themselves and things like markers are
implemented as actual Flutter widgets). A few caveats: first, it does not
use the Google Maps API; second, it is currently experimental (and not
feature complete) but is working. This is a radical new way of implementing
maps that I find exciting (like all Flutter widgets, it is highly
customizable), but of course there are risks and it is currently rather
immature. If you are interested, here's a link to their repository:

https://github.com/apptreesoftware/flutter_map

**Comment by feinstein** [comment]
As many here I am fairly new to Flutter and the lack of Google Maps support is a no go for my app to be migrated to Flutter.

That being said, please don't take me wrong, I understand the struggle on what I am about to suggest, but I believe Flutter should implement a Google Maps Widget made from the ground up, in Dart, with all the Google Maps APIs to support its power, just like the Android one is. 

Maps are supposed to be fast and interactive, we expect to be able to draw paths on the maps, add custom pins and etc. Also, we want to programmatically get places search autocomplete, reverse-geocoding, geocoding, Longitude and Latitude distance calculations support. 

Trying to use the Native Map on the device will probably be too limited for a full mobile experience, so my suggestion (as hard as it is) is to focus on a  full Google Maps Widget + Google Maps API for Flutter/Dart.

**Comment by rbellens** [comment]
I think that for a modern looking maps sdk (like google maps or mapbox), you’ll need opengl and since this is not yet supported, it will be difficult to build it from the ground up in dart (I tried doing it with canvas drawing, but it was too slow). I think there are two alternatives: (1) a less modern looking map, with simple image tiles (see e.g. https://github.com/apptreesoftware/flutter_map ) and (2) native plugins that draw to a SurfaceTexture that can be used in flutter through a Texture component (probably the open source mapbox sdk’s can be used as a basis for this). 

It is my understanding that the flutter authors are working on a maps sdk. What approach are they taking? What is the timeline? Will it be worth waiting for this or should the community start working on this?

 

 

 

 

Van: feinstein <notifications@github.com>
Beantwoorden - Aan: flutter/flutter <reply@reply.github.com>
Datum: woensdag 28 februari 2018 om 21:31
Aan: flutter/flutter <flutter@noreply.github.com>
CC: rbellens <rik.bellens@appsup.be>, Manual <manual@noreply.github.com>
Onderwerp: Re: [flutter/flutter] Inline Maps support (#73)

 

As many here I am fairly new to Flutter and the lack of Google Maps support is a no go for my app to be migrated to Flutter.

That being said, please don't take me wrong, I understand the struggle on what I am about to suggest, but I believe Flutter should implement a Google Maps Widget made from the ground up, in Dart, with all the Google Maps APIs to support its power, just like the Android one is.

Maps are supposed to be fast and interactive, we expect to be able to draw paths on the maps, add custom pins and etc. Also, we want to programmatically get places search autocomplete, reverse-geocoding, geocoding, Longitude and Latitude distance calculations support.

Trying to use the Native Map on the device will probably be too limited for a full mobile experience, so my suggestion (as hard as it is) is to focus on a full Google Maps Widget + Google Maps API for Flutter/Dart.

**Comment by matejthetree** [comment]
@mravn-google This is great news

**Comment by gedw99** [comment]
@NathanaelA
That's really interesting.
I was wondering the same thing for VR and AR.  Getting a flutter view to composite with output from other renders is the gist from what I can see ?
I was looking at exactly same issue with flutter for desktop ! I have flutter window but also a view from a 3d opengl rendering system.

**Comment by johnpryan** [comment]
> native plugins that draw to a SurfaceTexture that can be used in flutter through a Texture component (probably the open source mapbox sdk’s can be used as a basis for this).

@rbellens are there any changes you can recommend for https://github.com/apptreesoftware/flutter_map ? It sounds like there's an opportunity to improve the performance

**Comment by ezrasandzerbell** [comment]
@mravn-google do you have an estimate for when this new flutter map feature will roll out? Also, when you say that it will not be possible to draw on top of the map, does that include simple functionality like pin dropping / address display?

**Comment by mravn-google** [comment]
@ezrasandzerbell We've been delayed in this work which is going to happen in the context of the `google_mobile_maps` plugin [repo](https://github.com/flutter/plugins/tree/master/packages/google_mobile_maps). I expect activity to pick up very shortly.

Our initial (limited) solution will be to use native map views as overlays, later to be replaced by texture streaming from the native map views, once the GoogleMap APIs support that. The main point of the plugin until that happens is to support the development of the Dart API for controlling maps. This includes the ability to place markers etc.

**Comment by ezrasandzerbell** [comment]
@mravn-google Thanks for the new feature details. I noticed that the google_mobile_maps plugin doesn't have a public issues tab, so I am following up here. We would greatly appreciate documentation with code samples when these new features roll out. 
 
I am personally interested in: 

1) How to constrain a google map to specific width and heights. The solutions that I've seen so far (AppTree, MapView) both seemed to be full screen views only. Documentation is sparse for correct implementation and common use cases. 

2) How to hook up a search tool like Google Places Autocomplete (https://pub.dartlang.org/packages/flutter_google_places_autocomplete) so that users can type in a location, autocomplete their data, submit, and see that position rendered on a map.

**Comment by mravn-google** [comment]
@ezrasandzerbell The plugin work will come with both documentation and examples. Please file issues on this (flutter/flutter) repo for feature requests on the plugin.

**Comment by AfzalivE** [comment]
@pinkfish I think you could call back from flutter perhaps and set the height of your native view as needed. It's all kinda tricky

**Comment by EasonPai** [comment]
@mravn-google  I have issue when trying `google_mobile_maps`. I can run `google_mobile_maps` example on android (by replace GOOGLE_MAPS_API_KEY in gradle.properties to my own key). 

But for iOS, when I run example I get errors:
```
Syncing files to device iPhone SE...
[VERBOSE-2:dart_error.cc(16)] Unhandled exception:
	MissingPluginException(No implementation found for method init on channel plugins.flutter.io/google_mobile_maps)
	#0      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:153:7)
	<asynchronous suspension>
	#1      _channel (package:google_mobile_maps/google_mobile_maps.dart:16:9)
	#2      _channel (package:google_mobile_maps/google_mobile_maps.dart:14:21)
	#3      _GoogleMapsPlatformOverlay.create (package:google_mobile_maps/google_mobile_maps.dart:94:25)
	#4      PlatformOverlayController.attachTo.<anonymous closure> (package:google_mobile_maps/platform_overlay.dart:72:21)
	#5      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
	#6      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:938:9)
	#7      _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:751:7)
	#8      Timer._createTimer.<anonymous closure> (dart:async/runtime/libtimer_patch.dart:21:15)
	#9      _Timer._runTimers (dart:isolate/runtime/libtimer_impl.dart:382:19)
	#10     _Timer._handleMessage (dart:isolate/runtime/libtimer_impl.dart:416:5)
...
```

Besides, I have almost zero knowledge on iOS development.  I have no idea where should I assign the GOOGLE_MAPS_API_KEY to?

**Comment by mravn-google** [comment]
@EasonPai iOS is not supported by the plugin yet. We'll add that in the coming weeks.

**Comment by ocolot** [comment]
Do you have a release date for this feature? 

I might need to go back to React Native for a client project otherwise :(

**Comment by brooth** [comment]
@ocolot Same here. I've got to start a map-based project shortly and really wanted to use flutter but because of this map situation, I'm going back to RN.

**Comment by sateesh-penumalla** [comment]
@brooth, I am struggling to make maps work in RN. If you have a perfect example, please post it.

**Comment by alegsm** [comment]
Just gonna leave this here. If your needs of map usage are limited to asking the user for an address I implemented a plugin that handles everything, from location permission handling to reverse geocoding. https://pub.dartlang.org/packages/location_picker

**Comment by nitneuq33** [comment]
hello alegsm, I'm very interested about your plugin but I don't understand the example... I'm newbi I just want to set longitude and latitude and visualise geolocalisation. Did anyone succeed? 
does anyone have a simple example? thanks evrybody

**Comment by alegsm** [comment]
@nitneuq33  yes you can do it with location_picker. It is intended to work as a prompt for things as asking the user to input a delivery address. You can look at the example project inside the repo. ill leave you a link
https://github.com/touwolf/location_picker/tree/master/example

**Comment by rolurq** [comment]
Is there any updates about this?

**Comment by haydenflinner** [comment]
What's wrong with `flutter_map`?

**Comment by piotr-tobolski** [comment]
`flutter_map` is good substitute but it also significantly worse than native map views. It is using tiles instead of vector maps and doesn't support Google Maps let alone Apple Maps. I am personally waiting for streaming native view contents into flutter hierarchy as it will allow to close a lot more gaps than this one (e.g. WebViews, other SDKs) as referenced here: https://github.com/flutter/flutter/issues/730#issuecomment-370274688

**Comment by haydenflinner** [comment]
[Move on marker press not working iOS](https://github.com/react-community/react-native-maps/issues/1374)
[Inconsistent Polygon Rendering](https://github.com/react-community/react-native-maps/issues/2025)
[Marker offset not working in Android](https://github.com/react-community/react-native-maps/issues/2173)
[(one user) works on android not on IOS](https://github.com/react-community/react-native-maps/issues/2199)
[Z index doesn't work on iOS](https://github.com/react-community/react-native-maps/issues/1671)
[Marker rotation doesn't work on iOS](https://github.com/react-community/react-native-maps/issues/2186)
[Urltemplate not changing iOS](https://github.com/react-community/react-native-maps/issues/2200)
[Animated callout problem in iOS](https://github.com/react-community/react-native-maps/issues/1062)
[Marker onSelect doesn't work on Android](https://github.com/react-community/react-native-maps/issues/114)
[Android onclick not accurate](https://github.com/react-community/react-native-maps/issues/1562)
[Image in custom callout not working Android](https://github.com/react-community/react-native-maps/issues/1870)

I dumped React Native because of how awful its map support is, and I didn't care at all about things working on one platform and not on another. I think everyone's experience would be much better if we just added vector maps support to `flutter_map` (I don't know if gmap/apple map integration is doable?), for the same reasons that xster wrote about in [Why Flutter doesn’t use OEM widgets](https://medium.com/flutter-io/why-flutter-doesnt-use-oem-widgets-94746e812510)

**Comment by wmleler** [comment]
Note that flutter_map is based on Leaflet (rewritten into Dart/Flutter),
and there is a plugin for Leaflet that supports vector maps. Someone would
need to rewrite it into Dart/Flutter, but then you could use mapbox's
vector maps.

**Comment by Hixie** [comment]
We consider this to be a high priority and are working on it. It is going to take a while as it is a quite complicated problem to solve properly. We apologise for the delay and ask for your patience.

**Comment by wmleler** [comment]
Mapbox (the company that provides Leaflet and a wide assortment of vector
and raster map tiles) appears to have vector maps working in Flutter. Right
now it is working on Android only, but since they are using the texture map
technique (same as what inline video is using) it should not be difficult
to get it working on iOS as well.

https://github.com/mapbox/flutter-mapbox-gl

I'm believe they are taking pull requests.

Good to see all these map solutions coming from the community.

**Comment by jposuna** [comment]
@mravn-google the google_mobile_maps repo seems to have disappeared; Do you have any info?

**Comment by levrik** [comment]
@jposuna flutter/plugins#518

**Comment by mit-mit** [comment]
Renamed the present bug to make it clearer that this tracks inline 'Google Maps' support.

**Comment by rolurq** [comment]
@mit-mit Just to be clear, that means that flutter will __not__ provide a way to render offline maps?

**Comment by feinstein** [comment]
@mit-mit any timeline or expected release date on that long term? I have a project coming in and I would like to make it in Flutter, but without Google maps support I will have to go native.(Java/Swift).

**Comment by mpiannucci** [comment]
@feinstein there is a working bare bones plugin in the Flutter/plugins repo under google_maps_flutter.

**Comment by PetrShypila** [comment]
@mit-mit Indeed, some long-term release date would be really useful. The whole idea of flutter looks amazing. Unfortunately without Maps I hardly see starting a project on a Flutter

**Comment by mit-mit** [comment]
Sorry, we have no public schedule we can communicate yet :-(.

As mentioned a few comments up, our latest progress is in the plugins repo, and it does currently hold a very early version with some documented deficiencies. You may want to take a look at that.

**Comment by nitneuq33000** [comment]
hello there is way to customize the toolbar in map_view? or add widgets (floatingactionbutton). https://pub.dartlang.org/packages/map_view

currently map_view opens a new window full screen with a slide to the left with a black toolbar. I would like to customize the page transistion , color toolbar and add widgets to interact without having to go back.

Finally, it would not be possible to have the dynamic map in a pop up box instead of a full screen? 

thank you

**Comment by mravn-google** [comment]
@nitneuq33000 The `map_view` plugin is not maintained by the Flutter team. Please open issues at https://github.com/apptreesoftware/flutter_google_map_view.

**Comment by bjornbjorn** [comment]
@mravn-google Is polyline support on the roadmap for this plugin?

(what I've seen so far in the /example folder looks great btw!)

**Comment by mravn-google** [comment]
@bjornbjorn Full API coverage is on the roadmap where meaningful. Prioritization is very much informed by user feedback. So please [create an issue](https://github.com/flutter/flutter/issues/new) with your feature requests. Then other people can upvote them and/or feel motivated to submit PRs ... and/or we'll get to work on it asap.

**Comment by bjornbjorn** [comment]
Yes, I'm also referring to `google_maps_flutter`- I created an issue here: https://github.com/flutter/flutter/issues/17832

**Comment by xclud** [comment]
@mravn-google I am doing a Map widget by my own in pure Dart code (See my post above). At the moment i am implementing only the features i need for my project, but I would make it better overtime.

I will be happy to create a PR and see my work merged into Flutter. I don't know how to start and i appreciate your help getting started.

**Comment by mravn-google** [comment]
@xclud There are many ways to deliver maps in the context of Flutter, and it therefore makes little sense to elevate one of them and put it into Flutter (= the Flutter framework). I also think it makes little sense to try and consolidate different maps packages whose approaches are very different.

The `google_maps_flutter` package will remain a plugin, separate from the Flutter framework. It is going to mirror the iOS/Android APIs for Google Maps pretty closely, and some operations will involve asynchronous platform messages to make changes to the native map view.

Your package, being a pure Dart solution, is likely to expose a very different API, because you don't need to mirror or call the existing GoogleMaps Android and iOS APIs, and you don't need asynchronous platform messages.

**Comment by wmleler** [comment]
There is no one best solution for maps. We have some customers who really
want the real Google maps. We have other customers who need highly
customized maps.

If you are interested in a map solution that is based on Flutter (where all
elements of the map are Flutter widgets), check out
https://github.com/apptreesoftware/flutter_map

**Comment by feinstein** [comment]
@wmleler I am sorry but I still don't understand, what's the problem with making "the real Google Maps" in Dart as part of Flutter? Is it there any limitations?

**Comment by MisterJimson** [comment]
I think we would want to avoid a dependency on a separate service/server in
Flutter core.

This is what plugins are made for, either official flutter supported
plugins or third party.

**Comment by mravn-google** [comment]
@feinstein Google Maps is just one maps provider (albeit an important one). But Flutter apps should also work with MapBox and any other maps provider out there.

Flutter is not about consolidating everything into a monolithic code base. That's why we have packages and plugins separate from the Flutter framework.

That said, in the ideal world, the Google Maps team would supply a pure Dart maps widget as a Flutter package. MapBox would supply their take on the same thing. And so on. This is bound to happen, if Flutter becomes popular enough. In the meantime, we need to integrate the maps components we have.

**Comment by feinstein** [comment]
@mravn-google thanks, this makes perfect sense to me! 

On that topic, I think Flutter will be adopted a lot quicker if there was a supporting ecosystem around it, the same as Java for Android currently has, that being Dagger 2, Retrofit, Picasso, Room, RxAndroid, Junit, Expresso etc... (I know some of this functionalities we can already find in Flutter, but I just wanted to be thorough on my list nonetheless). I understand this is asking a lot of you guys, but I can see a speed up on adptoion once developers see an easy migration and don't miss anything from the old platform.

**Comment by Hixie** [comment]
@feinstein Do you mean something like https://pub.dartlang.org/flutter ?

**Comment by feinstein** [comment]
@Hixie some time a go I made this reddit post:

https://www.reddit.com/r/FlutterDev/comments/839j38/dartflutter_ecosystem/?utm_source=reddit-android

And the reply I got is that DI, REST(Retrofit) and JSON (GSON, Jackson) isn't available yet... Did it change?

**Comment by Hixie** [comment]
@feinstein Let's move this discussion to separate bugs. Can you file a bug for each package you think is missing, and we'll see what we can do to address your need in each case? Thanks.

**Comment by feinstein** [comment]
@Hixie, sure thing!

**Comment by jposuna** [comment]
I've been playing around with the plugin, and I'm not able to do the following:
I want to be able to navigate to a page that contains the map widget, in full screen, and when pressing "back" (or other actions), the map widget disappears.

From what I can see, the examples folder attaches the navigatorObservers when creating the app. My issue is that GoogleMapOverlayController needs a width and a height to be created, and the only way I've been able to get the dimensions of the screen is using a LayoutBuilder (getting the BoxConstraints), at build stage (ie. can't do it statically). To get the map to be interactive, I need to manually attach it to the context. But then there's no way I've been able to detach it. When I press the back button, the app navigates back, but the map is still over all the contents.

Below is the code I've been trying.

**Page 1**
`Navigator.push(context, new MaterialPageRoute(builder: (context) => new GoogleMapsPage()));`


**Google Maps Page**
```
class GoogleMapsPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(title: const Text("Test Maps")),
      body: new GoogleMaps(),
    );
  }
}

class GoogleMaps extends StatefulWidget {
  GoogleMaps({Key key}) : super(key: key);

  @override
  _GoogleMaps createState() => new _GoogleMaps();
}

class _GoogleMaps extends State<GoogleMaps> {

  GoogleMapOverlayController mapOverlayController;

  @override
  Widget build(BuildContext context) {
    return new Center(
      child: LayoutBuilder(
          builder: (BuildContext context, BoxConstraints constraints) {
        mapOverlayController = GoogleMapOverlayController.fromSize(
          width: constraints.maxWidth,
          height: constraints.maxHeight,
          options: GoogleMapOptions(
            cameraPosition: const CameraPosition(
              target: LatLng(-34.022631, 150.620685),
              zoom: 15.0,
            ),
            tiltGesturesEnabled: false,
            rotateGesturesEnabled: false,
          ),
        );
        mapOverlayController.overlayController.activateOverlay();
        mapOverlayController.overlayController.attachTo(context);
        return GoogleMapOverlay(controller: mapOverlayController);
      }),
    );
  }

  @override
  void deactivate() {
    super.deactivate();
    mapOverlayController.overlayController.deactivateOverlay();
    mapOverlayController.overlayController.detach();
    mapOverlayController.overlayController.dispose();
    mapOverlayController.mapController.dispose();
    mapOverlayController.dispose();
  }
}
```

Any help would be appreciated.

Thanks!

**Comment by jposuna** [comment]
@mravn-google thanks for the quick reply. To be clear, the limitation I'm running into is not full-screen support (I was able to get it full-screen), but getting it off the screen when navigating away.

I'm completely new to flutter, so there might be jargon that is escaping me, but from the README ("Issues and pull requests aimed to help us prioritize and speed up this effort are very welcome"), and your earlier comment on this thread ("Full API coverage is on the roadmap where meaningful. Prioritization is very much informed by user feedback") I understood that it was a preview meant to be expanded over time based on use-cases and feedback.

But from your answer, it seems like it's not meant to develop into a fully-fledged solution, but rather serve as a proof-of-concept before the real solution. I'm I understanding it correctly? Either way is fine, I'm just trying to understanding the situation, before doing a full integration with one solution, and then having to change the whole integration down the line.

**Comment by mravn-google** [comment]
@jposuna We intend to expand the `GoogleMapController` part of the plugin to a full solution, including 1) full coverage of relevant Google Maps APIs for controlling what is shown in the map view and responding to user interaction with that view, and 2) integrating the map view as a standard widget, to be freely composed with other widgets, including those implementing navigation.

The `GoogleMapOverlayController`, and platform overlays in general, is a crutch for doing 2), a temporary solution because we currently don't have the Google Maps APIs needed for the real widget integration. We do not intend to work on the crutch, because it cannot carry us to where we want to go.

This means that the part of your code that depends on `GoogleMapOverlayController` will eventually have to change. And in case your code must employ hacks to work around the limitations of platform overlays, those hacks will eventually become unnecessary.

**Comment by tieorange** [comment]
Our company can't wait for google maps support on Flutter

**Comment by levrik** [comment]
@tieorange Then I would suggest to pick another framework or build native apps.

**Comment by kitty7756** [comment]
Also waiting for this support 👍

**Comment by PauloMelo** [comment]
I'm having problems compiling, I get "ld: framework not found google_maps_flutter "
because of another plugin I have use_frameworks! on my PodFile. 

My flutter doctor.
[✓] Flutter (Channel master, v0.5.2-pre.55, on Mac OS X 10.13.4 17E202, locale en-PT)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[✓] iOS toolchain - develop for iOS devices (Xcode 9.4)
[✓] Android Studio (version 3.1)
[✓] VS Code (version 1.23.1)
[✓] Connected devices (3 available)

**Comment by slightfoot** [comment]
@PauloMelo run `flutter build ios` from your project directory. Then xcode will build fine.

**Comment by MrVuLong** [comment]
@PauloMelo Have you add 
`google_maps_flutter:
   git:
     url: git://github.com/flutter/plugins
     path: packages/google_maps_flutter`
as a dependency in your pubspec.yaml file?

**Comment by mravn-google** [comment]
Seems the `google_maps_flutter.podspec` file needs a
```Ruby
s.static_framework = true
```
line. I'll cook up a PR.

**Comment by PauloMelo** [comment]
@mravn-google Thanks, that solved my problem.

**Comment by Georgian** [comment]
Started a new flutter project which should include a map view, and I hit this brick wall. Waiting for the good news!

**Comment by gerryhigh** [comment]
@jposuna I have gotten it to disappear when navigating away. When I get a chance I'll post what I'm doing. Here's a short video of it in action: https://twitter.com/gerryhigh/status/997312139663757312

**Comment by nitneuq33** [comment]
hello, I just tested the plugin and it's really impressive ! I like the
controlleur and all customisations and possibilities. I just see somes bug.
like crash after click on the  newCameraPosition  button and I pinch to
zoom from the back. other experience, I tried to integer map in a Card to
have ronded border, but it's doesn,'t work. other I tried to make a
geofencing zone, I stack a circular shape with 50% opacity over the map but
impossible to set le circular shape over the map.

2018-06-11 21:58 GMT+02:00 Gerry High <notifications@github.com>:

**Comment by piotr-tobolski** [comment]
@nitneuq33 As you can read in the README.md of this plugin - it works by placing platform view as overlay over the flutter view and it doesn't allow freely compositing with other widgets.

**Comment by nitneuq33** [comment]
Sorry OK,is there any possibilities to implement geofencing controller ?
like a custom tag but circular and dynamic in function of the zoom and
value of the radius ?

2018-06-14 12:49 GMT+02:00 Piotr Tobolski <notifications@github.com>:

**Comment by daldoy** [comment]
Is there an ETA for this feature?

**Comment by nitneuq33** [comment]
Sorry what is ETA ?

**Comment by feinstein** [comment]
Estimated Time of Arrival.

And no, there isn't, have been asked many times before already.

**Comment by nitneuq33** [comment]
hello, I tried to use the exemple
https://github.com/flutter/plugins/tree/master/packages/google_maps_flutter/example/lib
but with simple view, I tried to display map after app launch with only 3
button ( sattelite type, hybrid type and new location ). I succes to do
this but without change the page architecture of the example. so I tried to
use the other simpler example
https://github.com/flutter/plugins/tree/master/packages/google_maps_flutter
but it use StatLessWidget and the exemple of map_ui use statfulWidget but
when I tried to use statfullwidget I have errors. Can I have a simple
example with statfullWidget to use map at the launch of the app ? ( just
need of 3 button to change the type of map and lanch a new localisation ) .
I'm new on flutter sorry if it's not clear but this plugin is realy
important and this is the last bloc for me :)

2018-06-14 23:03 GMT+02:00 feinstein <notifications@github.com>:

**Comment by mit-mit** [comment]
@nitneuq33 sorry to hear you are having issues. For support questions like that, you are likely going to get better support on stackoverflow (https://stackoverflow.com/questions/tagged/flutter). The present issue tracker is meant to track overall feature progress.

**Comment by nitneuq33000** [comment]
I tried to explain my need in stackoverflow but I have not enought experience to debug and understand the answers. Anyone to write simple exemple of his plugin with only map on home screen compatible with option like marker, mapType etc. ? thank you

**Comment by gerryhigh** [comment]
@nitneuq33000 Here's a simple example using the plugin with markers, info windows (and callback when info window tapped) & driving directions. https://github.com/gerryhigh/Flutter-Google-Maps-Demo

@@jposuna take a look as the map disappears when navigating away

Note that this example has a button on a home page which navigates to the map page but it would be trivial for you to make the venues page the home page. This example also points to a fork of the plugins in pubspec.yaml because it has a fix for the infowindowonTapped call (I need create a pull request for the fix).

Lastly, this approach does not use the navigator observer that the maps demo that comes with the plugin uses. As a result I've noticed that there is a bug on iOS when rendering the map where the overlay will show up and some x offset. Currently I'm adding a delay on load to get around this issue.

To run this example just input your map keys. I hope this helps.

**Comment by nitneuq33** [comment]
thank you it's very helpful, really what I searched :)

2018-06-16 21:41 GMT+02:00 Gerry High <notifications@github.com>:

**Comment by nitneuq33** [comment]
I have a bug with

initState

I can't upload a saved value. I use shared preference to pass data to
the next page, but when I add this line, I have un bug and the map is
always active even when I back to the previous page.

void initState() {
  super.initState();

_loadgeoValue()

}

...

_loadgeoValue() async {

  SharedPreferences prefs = await SharedPreferences.getInstance();
  setState(() {
    lat = (prefs.getDouble('lat1'));
    long = (prefs.getDouble('long2'));
  }
  );
}

If some one have an idea... thanks for advance



2018-06-17 10:22 GMT+02:00 quentin guichot <guichotquentin@gmail.com>:

**Comment by nitneuq33** [comment]
Hello, I succeded to get around without passing into initstate to upload my
data when I launch the new page. So currently it's fonctionnal but the
fixed layer when I close the page isn't very sexy. Any idea to make it more
visual, like a fade out animation of the layermap when you close the page ?
or transition animation  who catch the transition of the page ? thank you

**Comment by cosminstoinescu** [comment]
Hello, is there any progress regarding this plugin, any news on when we shoul expect to have a solutin?
Thank you!

**Comment by Stavarmath** [comment]
How to get current user location lat/long with this plugin? is it supported?

**Comment by hundeva** [comment]
Please keep the support questions away from this thread, ask them on Gitter, Stackoverflow, or any other relevant forum. Anytime you comment, a lot of people get an unnecessary notification. This thread should only contain relevant status updates, or actually useful discussion, thanks.

**Comment by NguyenSen** [comment]
How to use other widget on  this plugin. It is hard to insert widget on map.

**Comment by fritz-flaconi** [comment]
@NguyenSen it says in the README that it can't be combined with other plugins

**Comment by fritz-flaconi** [comment]
Hi. I assume using a WebView  to embed the web version of Google Maps is totally unrealistic? (Haven't tried yet, but I have just noticed this plugin https://pub.dartlang.org/packages/flutter_webview_plugin

**Comment by cosminstoinescu** [comment]
Hello, is there any progress regarding this plugin, any news on when we shoul expect to have a solutin?
Thank you!

**Comment by mclark4386** [comment]
I'm curious if the concept of "native widgets" still progressing. They were going to let you wrap a "native" view in a flutter widget in a plugin to let you, theoretically, make a mapview plugin that just wrapped the "native" views and inserted that in the flutter render tree where ever you had used the plugin's widget.

**Comment by amirh** [comment]
I'm actively working on this, unfortunately I don't have an ETA.
You can also follow #19030 for more detailed progress updates.

**Comment by mclark4386** [comment]
@amirh subscribed thank you!

**Comment by thiphariel** [comment]
Maybe you should take a look here : https://github.com/apptreesoftware/flutter_google_map_view

@apptreesoftware

**Comment by amirh** [comment]
An Android only **preview** of the inline google maps plugin is now available at: https://github.com/flutter/plugins/tree/master/packages/google_maps_flutter

(the framework and engine support required for this is currently only available on the master channel)

**Comment by feinstein** [comment]
The Readme speaks about iOS setup ... Shouldn't this be more clarified?

**Comment by amirh** [comment]
Good point, I'll remove it for now.

**Comment by cosminstoinescu** [comment]
Hi, when testing google_maps_flutter I encountered the following problem:
If I press 'Go to London' button, the map goes at that location, and if I try to do a zoom out the application stop working and closes.
Log:
E/JavaBinder(10217): !!! FAILED BINDER TRANSACTION !!!  (parcel size = 3050800)
I/Process (10217): Sending signal. PID: 10217 SIG: 9
Lost connection to device.

If I do a zoom in, everithing works, and then if I do a zoom out, the aplication works fine until it reaches the zoom value set when the button was pressed.
If I dont press the button, I can do zoom in and zoo out without any problems.

**Comment by zoechi** [comment]
@cosminstoinescu it's probably better to create a new issue for this specific problem with a completely filled issue template and if possible a running minimal reproduction.

**Comment by cosminstoinescu** [comment]
Created issue ticket: #22228: Inline Google Maps - zoom crash

**Comment by syfgkjasdkn** [comment]
As others have already mentioned, it's possible for flutter to interact with native maps with platform channels, here's an example of doing just that in swift: https://github.com/syfgkjasdkn/map_and_flutter

Just in case someone might find it helpful.

**Comment by feinstein** [comment]
Out of curiosity, are you guys also porting the Places API, or just the Google Maps API?

**Comment by HerrNiklasRaab** [comment]
Is this the right thread for getting notified on any updates, i need inline google maps view for android and ios, is there a more detailed thread or something like this?

**Comment by MisterJimson** [comment]
@HerrNiklasRaab yes this is the issue to watch and has the best information for inline maps.

**Comment by amirh** [comment]
flutter/plugins#892 added iOS support to the google_maps_flutter plugin.
The platform views embedding support, which the inline map is based on, is still in early preview (see https://github.com/flutter/flutter/issues/19030#issuecomment-437534853 ).

Open issues are tagged with the [platform-views](https://github.com/flutter/flutter/labels/a%3A%20platform-views) and/or [maps](https://github.com/flutter/flutter/labels/p%3A%20maps) labels, please upvote the issues you care about to help us prioritize.

**Comment by amirh** [comment]
Note that if you want to try it out on iOS you will need to enable the preview flag, see more details on:
https://github.com/flutter/flutter/issues/19030#issuecomment-437534853

**Comment by feinstein** [comment]
Ohh, that's great! I remember reading there couldn't be any widgets on top of the map, as if it was a different rendering layer. Is this valid for ios and android?

**Comment by nosmirck** [comment]
After trying the maps plugin and messing around with platform views widgets, I came to this post, the very original post, and I can say that option 3 would have been a much better option.

I'd like to know, is there a reason why no one chose to create an actual widget for maps from scratch? I mean, the whole power for it is there! Skia is powerful enough to create it's own renderer for buildings and roads using simple graphics, or is it not possible to do so from Dart? I checked the code from rendering a simple box from Dart to the engine and I see it's possible.

I understand there's a technical reason we need platform views, but, maps needs "native" flutter SDK, or what's the plan for Fuchsia? Keep leveraging Android's SDK to render maps? It's nonsense!

We already have a platform view that seems to be working fine, might need optimizations, but the teams can now just drop this maps implementation (basically two, one per platform) and start working on an actual maps for Flutter that would technically work on any future platform that flutter gives support, including hummingbird, PC, Mac and Linux (and many more).

It's the best way, and you now it. The maps plugin for Unity3D doesn't use the native view, it's entirely custom for the engine and with amazing customizability! We need that for Flutter too, it's much more rich and useful than what a platform view of a map can do.

Also, it will bring an idea for us, developers, on how actually use the graphics engine on a low level to build other stuff.

I hope you guys don't take this wrong, but, I truly believe that the maps plugin through platform views already fulfilled his goal: to make possible native views to be rendered as widgets.

Oh, before I end this: I thought that maybe this could extend to WebViews (having an actual browser engine that renders using skia) is something useful, maybe not, maybe it's just too much (for now) but would solve the issue of having different renderers for WebViews per platform (like the intention of the flutter maps), however, an integration with chrome engine somehow would be amazing, I mean, if the platform is able of running chromium it could technically serve as a web renderer and provide a channel to render WebViews inside Flutter.

That's it! I'm going to bed now :)

**Comment by hundeva** [comment]
I assume that the very long time plan is to develop something independent from the underlying platform (option 3). However, in the meanwhile, this was a "long hanging fruit", to support the current two main platforms. I have no doubt that before long, third party solutions will pop up, for example nothing stops noone from creating an OSM package.
